### PR TITLE
`Exam mode`: Update exam cards when pressing the refresh button

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
During the exam mode testing sessions we noticed that the refresh button has no impact on the exam cards. This is especially noticeable when waiting for an exam to get visible

### Description
Update the exam cards when a course update gets retrived.

### Steps for Testing
Prerequisites:
- 2 Student
- 1 Exam with visibility date in the future

1. Open the exam cards page -> exam is not yet visible
2. Wait unit after the visibility date and press refresh -> exam gets visible

#### Performance Review
- [x] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by JohannesStoehr — APPROVED** [review]
Code changes look good to me 👍

**Review by laadvo — DISMISSED** [review]
Tested on ts1, the exam cards get updated upon refreshing

File: src/main/webapp/app/overview/course-exams/course-exams.component.ts [hunk 1]
```diff
@@ -59,17 +60,25 @@ export class CourseExamsComponent implements OnInit, OnDestroy {
...
+            this.expandAttemptsMap = new Map(this.course.exams.filter((exam) => exam.testExam && this.isVisible(exam)).map((exam) => [exam.id!, false]));
+            this.updateExams();
+        }
+    }
 
+    private updateExams(): void {
+        if (this.course?.exams) {
             // Loading the exams from the course
             const exams = this.course.exams.filter((exam) => this.isVisible(exam)).sort((se1, se2) => this.sortExamsByStartDate(se1, se2));
+            // add new exams tho the attempt map
```
Comment (Strohgelaender) [inline src/main/webapp/app/overview/course-exams/course-exams.component.ts:72]:
```suggestion
            // add new exams to the attempt map
```

**Review by laadvo — APPROVED** [review]
Reapprove test after last commit

**Review by tobias-lippert — APPROVED** [review]
Code LGTM

**Review by jakubriegel — APPROVED** [review]
Works on ts1 👍🏻

**Review by tobias-lippert — APPROVED** [review]
Tested on ts1. The refresh button works again.

**Review by nityanandaz — APPROVED** [review]
I confirm the card did update (manually tested)

**Pull request merged**